covid/question_covid.py

```diff
@@ -51,20 +60,68 @@ def get_answer(self, text, question):
 
         return answer, score
 
-    def predict(self, question):
+    def rank_shortlisted(self, question, list_of_texts):
+        """ Ranks question/document pairs based on scibert"""
+
+        # Load scibert
+
+        embedding_question = self._scibert_embedding(question)
+
+        similarities = [(i,
+                         tf_idf_score,
+                         cosine_similarity(embedding_question, self._scibert_embedding(text))[0, ],
+                         text)
+                        for i, tf_idf_score, text in list_of_texts]
+        similarities = sorted(similarities, key=lambda x: x[2], reverse=True)
+
+        return similarities
+
+    def _scibert_embedding(self, x):
+        # Max sequence length is 512 for BERT
+        if len(x) > 512:
+            embedded_a = self._scibert_embedding(x[:512])
+            embedded_b = self._scibert_embedding(x[512:])
```

Comment:
so the second part of the text is very likely to be <512 in length?

Reply:
No, if the second part is bigger again, it will call the function recursively again

Reply:
@nsorros's masterpiece 🤓

Reply:
far from masterpiece, haha

Reply:
oo that's cool. function inception